exla/c_src/exla/exla_client.cc

```diff
@@ -177,7 +177,8 @@ xla::StatusOr<ExlaBuffer*> ExlaClient::BufferFromBinary(const ErlNifBinary& bina
   xla::PjRtClient::HostBufferSemantics semantics = xla::PjRtClient::HostBufferSemantics::kImmutableUntilTransferCompletes;
 
   EXLA_ASSIGN_OR_RETURN(xla::PjRtDevice* device, client_->LookupDevice(device_id));
-  EXLA_ASSIGN_OR_RETURN(auto buffer, client_->BufferFromHostBuffer(binary.data, shape, semantics, nullptr, device));
+  EXLA_ASSIGN_OR_RETURN(auto buffer, client_->BufferFromHostBuffer(
+    binary.data, shape.element_type(), shape.dimensions(), absl::nullopt, semantics, nullptr, device));
```

Comment:
The corresponding XLA change: https://github.com/tensorflow/tensorflow/commit/59077ac39cc64139f889863081bdb3d0f88d39b6

The `absl::nullopt` is for strides, which I don't think we need. The referenced commit message explains that NumPy keeps strides information on tensors, which allows transposition without any changes to the buffer. They now allow passing it to this function, so the buffer translation doesn't happen on the Python side.